src/tiny_slam/tiny_scan_cost_estimator.h

```diff
@@ -3,8 +3,24 @@
 
 #include "../core/grid_scan_matcher.h"
 
+/*!
+ * \brief An implementation of the scan cost function described in the original tinySLAM paper.
+ *
+ * This class presents one method to calculate a numeric value of one frame. This method is defined in the article.
+ */
 class TinyScanCostEstimator : public ScanCostEstimator {
 public:
+
+  /*!
+   * Calculates a discrepancy between a given scan and a map assuming a given robot position using the approach described in the tinySLAM paper
+   * The cost is calculated by summing up probabilities of being empty for the cells that are assumed to be occupied by laser scan.
+   * The lower the value the more probable the robot pose.
+   * \param[in] pose     - the robot pose in the space
+   * \param[in] scan     - array of points given from the laser scanner when robot was located in one place
+   * \param[in] map      - the environment map built on the previous steps
+   * \param[in] min_cost - the cost value which presented as a limit after that there will be no reason to calculate this scan
```

Comment:
Make it concise. Consider: "The minimum found  cost for the scan at the moment."